File: macos/QMK Toolbox/USB/Bootloader/AtmelSAMBADevice.m
#import "AtmelSAMBADevice.h"

@implementation AtmelSAMBADevice

- (id)initWithUSBDevice:(USBDevice *)usbDevice {
    if (self = [super initWithUSBDevice:usbDevice]) {
        self.name = @"Atmel SAM-BA";
        self.type = BootloaderTypeAtmelSamBa;
        self.resettable = YES;
        while (self.serialPort == nil) {
            self.serialPort = [self findSerialPort];
        }
    }
    return self;
}

-(void)flashWithMCU:(NSString *)mcu file:(NSString *)file {
    [self runProcess:@"mdloader" withArgs:@[@"-p", self.serialPort, @"-D", file, @"--restart"]];
}

-(void)resetWithMCU:(NSString *)mcu {
    [self runProcess:@"mdloader" withArgs:@[@"-p", self.serialPort, @"--restart"]];
}

- (NSString *)description {
    return [NSString stringWithFormat:@"%@ [%@]", [super description], self.serialPort];
}

@end

File: macos/QMK Toolbox/USB/Bootloader/AtmelSAMBADevice.swift
import Foundation

class AtmelSAMBADevice: BootloaderDevice {
    private var serialPort: String?

    override init(usbDevice: USBDevice) {
        super.init(usbDevice: usbDevice)
        name = "Atmel SAM-BA"
        type = .atmelSamBa
        resettable = true
        while serialPort == nil {
            serialPort = findSerialPort()
        }
    }

    override func flash(_ mcu: String, file: String) {
        runProcess("mdloader", args: ["-p", serialPort!, "-D", file, "--restart"])
    }

    override func reset(_ mcu: String) {
        runProcess("mdloader", args: ["-p", serialPort!, "--restart"])
    }

    override var description: String {
        "\(super.description) [\(serialPort!)]"
    }
}
